conans/test/functional/toolchain/test_adjust_auto.py

```diff
@@ -435,8 +437,11 @@ def test_fPIC(self, fpic):
         self.assertIn("-- Conan: Adjusting fPIC flag ({})".format(fpic_str), configure_out)
         self.assertIn(">> CMAKE_POSITION_INDEPENDENT_CODE: {}".format(fpic_str), configure_out)
 
-        type_str = "STRING" if self.use_toolchain else "UNINITIALIZED"
-        self.assertEqual(fpic_str, cmake_cache["CONAN_CMAKE_POSITION_INDEPENDENT_CODE:" + type_str])
+        if self.use_toolchain:
+            self.assertNotIn("CONAN_CMAKE_POSITION_INDEPENDENT_CODE", cmake_cache_keys)
```

Comment:
I guess that we should check for ``CMAKE_POSITION_INDEPENDENT_CODE`` now instead? The idea is that the conan variable does not escape the toolchain, but the real CMake configuration is actually applied.

Reply:
No. Even if this variable is not set in the CMakeCache, it is available with the proper value while running `configure` (the test is checking two lines before the actual value printed in the `CMakeLists.txt`) and CMake is adding the flags to the build-system files it generates.


-- 
IMO `CMakeCache` is a little bit misleading.

When using the GUI for CMake you run two steps: `configure` and `generate`, from the command line only one of them `cmake . -G "..."`. The CMake-GUI `configure` step generates the `CMakeCache.txt` file and `generate` uses it (only that file?) to generate the files for the specific build-system. Once the files for VisualStudio, make,... are generated the values contained in the `CMakeCache.txt` are no longer used (unless you want to edit them and run `generate` again... removing existing files before).

Do we want to support the `generate` step without running `configure` before? Is it just running `cmake . ` without using `-DCMAKE_TOOLCHAIN_FILE`? Then we would need to store everything from the toolchain in the `CMakeCache` and 🤞 . I suppose it would work (but I'd need to intensively test it). Is this a workflow we want to support (investigate and ensure it works)?